### PR TITLE
Disable analytics and default report submissions

### PR DESCRIPTION
**Pull request by johnmccabe**
This PR removes all analytics (Google) from the Lumogon client, and also disables the automatic submission of harvested reports to the Puppet managed reporter backend (via the `report` subcommand) in advance of that service being shut down.

The report command has been updated to require an endpoint to be set, either via the `--endpoint` argument or via the `LUMOGON_ENDPOINT` environment variable.

Documentation has been updated accordingly, the ANALYTICS.md file has been left for reference with a note indicating the data is no longer gathered.

### CODE REVIEW

**Comment by johnmccabe** [comment]
@scottyw could I get 👀 on this please, pending Travis going green.

File: Makefile [hunk 1]
```diff
@@ -54,19 +46,16 @@ todo:
...
+	golint ./...
 
 vet: bootstrap
-	go vet `glide novendor`
+	go vet ./...
 
 licenses: $(GOPATH)/bin/licenses
 	@licenses  $(PACKAGE_NAME) | grep $(PACKAGE_NAME)/vendor
 
 all: clean vendor test build image puppet-module
```
Comment (scottyw) [inline Makefile]: It looks like the `vendor` target was removed?

What about also running `dep` as part of the build?

**Comment by Philip-Smyth** [comment]
GTG!
catbug its fine

**Comment by johnmccabe** [comment]
Thanks @Philip-Smyth !

**Pull request merged**